client/src/legacy/ElementEditor/entwine.js

```diff
@@ -60,16 +62,35 @@ jQuery.entwine('ss', ($) => {
     },
 
     onunmatch() {
-      resetStores();
+      // Reset the store if the user navigates to a different part of the CMS
+      // or after submission if there are no validation errors
+      if (!$('.cms-edit-form').data('hasValidationErrors')) {
+        resetStores();
+      }
```

Comment:
The condition doesn't explicitly test for anything to see if it's swapping to a new page - but it doesn't need to. This condition evaluates to `true` when navigating to a new page. I've confirmed that this is true under the following scenarios:
1. No form submission has happened yet. I click to a different page.
2. A form submission happened and there were no validation errors. I click to a different page.
3. A form submission happened and there _were_ validation errors. I click to a different page.